File: Python/Mycar.py
class My_car:

    def __init__(self, color, model, old_car_value, discount, new_price, new_ride):
        self.color = color
        self.model = model
        self.old_car_value = old_car_value
        self.discount = discount
        self.new_price = new_price
        self.new_ride = new_ride

    def my_current_car(self):
        return "My Golf"
    
    def mycar_4_sale(self):
        return "for sale"

    def my_sold_car(self):
        return "Sold!"

    def my_new_car(self, new_ride, model, color):
        return "My new vehicle will be a "+ new_ride + model + color

    def new_car_price(self, new_price, discount):
        precio_total = new_price - discount
        return precio_total

    def price_with_discount(self, new_price, old_car_value):
        precio_descuento = new_price - old_car_value
        return precio_descuento

new_car = My_car("Gris", " 2005 ", 40000, 5000, 150000, "Hilux")#aqui estamos instanciando un objecto de la clase my_car el cual lleva un constructor
print(new_car.my_current_car())
print(new_car.mycar_4_sale())
print(new_car.my_sold_car())
print(new_car.my_new_car(new_car.new_ride, new_car.model, new_car.color))
print("price with 5k discount ", new_car.new_car_price(new_car.new_price, new_car.discount))
print("with no discount but old car value: ", new_car.price_with_discount(new_car.new_price, new_car.old_car_value))

File: Java/classes/Mycar.java
package Classes;

public class Mycar {
    public String color;
    public String model;
    public double old_car_value;
    public double discount;
    public double new_price;
    public String new_ride;

    public Mycar(String color, String model, double old_car_value, double discount, int new_price, String new_ride){
        this.color = color;
        this.model = model;
        this.old_car_value = old_car_value;
        this.discount = discount;
        this.new_price = new_price;
        this.new_ride = new_ride;
    }

    public void my_current_car(){
        System.out.println("My Golf");
    }
    public void mycar_4_sale(){
        System.out.println("Car for sale");
    }

    public void my_sold_car(){
        System.out.println("Sold!");
    }

    public void my_new_car(String new_ride, String model, String color){
        System.out.println ("My new vehicle will be: "+new_ride + model + color);
    }
    
    public void new_car_price(double new_price, double discount){
        double precio_total = this.new_price - this.discount;
        System.out.println("price with 5k discount " + precio_total);
    }
    
    public void new_car_with_discount(double new_price, double old_car_value){
        double precio_descuento = this.new_price - this.old_car_value;
        System.out.println("with no discount but old car value: " + precio_descuento);
    }

}


